nodebuilder/tests/tastora/framework.go

```diff
@@ -0,0 +1,417 @@
+package tastora
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/moby/moby/client"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/celestiaorg/celestia-app/v4/app"
+	tastoradockertypes "github.com/celestiaorg/tastora/framework/docker"
+	"github.com/celestiaorg/tastora/framework/testutil/sdkacc"
+	"github.com/celestiaorg/tastora/framework/testutil/toml"
+	"github.com/celestiaorg/tastora/framework/testutil/wait"
+	"github.com/celestiaorg/tastora/framework/testutil/wallet"
+	tastoratypes "github.com/celestiaorg/tastora/framework/types"
+
+	rpcclient "github.com/celestiaorg/celestia-node/api/rpc/client"
+)
+
+const (
+	celestiaAppImage   = "ghcr.io/celestiaorg/celestia-app"
+	defaultCelestiaTag = "v4.0.0-rc6"
+	nodeImage          = "ghcr.io/celestiaorg/celestia-node"
+	defaultNodeTag     = "v0.23.2-mocha"
+	testChainID        = "test"
+)
+
+// DefaultTestTimeout should be used as the default timeout on all Tastora tests.
+// It's generously set to 10 minutes to give enough time for CI.
+const DefaultTestTimeout = 10 * time.Minute
+
+// Framework represents the main testing infrastructure for Tastora-based tests.
+// It provides Docker-based chain and node setup, similar to how Swamp provides
+// mock-based testing infrastructure.
+type Framework struct {
+	t       *testing.T
+	logger  *zap.Logger
+	client  *client.Client
+	network string
+
+	provider   tastoratypes.Provider
+	daNetwork  tastoratypes.DataAvailabilityNetwork
+	bridgeNode tastoratypes.DANode
+	fullNodes  []tastoratypes.DANode
+	lightNodes []tastoratypes.DANode
+	celestia   tastoratypes.Chain
+}
+
+// NewFramework creates a new Tastora testing framework instance.
+// Similar to swamp.NewSwamp(), this sets up the complete testing environment.
+func NewFramework(t *testing.T, options ...Option) *Framework {
+	f := &Framework{
+		t:      t,
+		logger: zaptest.NewLogger(t),
+	}
+
+	// Apply configuration options
+	cfg := defaultConfig()
+	for _, opt := range options {
+		opt(cfg)
+	}
+
+	f.logger.Info("Setting up Tastora framework", zap.String("test", t.Name()))
+	f.client, f.network = tastoradockertypes.DockerSetup(t)
+	f.provider = f.createDockerProvider(cfg)
+
+	// Cleanup function
+	t.Cleanup(f.cleanup)
+
+	return f
+}
+
+// SetupNetwork initializes the complete network infrastructure.
+// This includes starting the Celestia chain and all DA nodes.
+func (f *Framework) SetupNetwork(ctx context.Context) error {
```

Comment:
just an observation, it looks like the `Framework` type is implementing a lot of lifecycle type functions that exist in `suite.Suite`

did you consider making `Framework` a `suite.Suite` and implementing these as first class `SetupTest` / `SetupSuite` type functions?

```go
type Framework struct {
  suite.Suite
  // ....
}
```

would give access to all the `f.T().Require()` fns as an added bonus.